src/Data/Digit.agda

```diff
@@ -55,11 +55,8 @@ toNatDigits base@(suc (suc _)) n = aux (<-wellFounded-fast n) []
   aux : {n : ℕ} → Acc _<_ n → List ℕ → List ℕ
   aux {zero}        _      xs =  (0 ∷ xs)
   aux {n@(suc _)} (acc wf) xs with does (0 <? n / base)
-  ... | false = (n % base) ∷ xs
-  ... | true  = aux (wf q<n) ((n % base) ∷ xs)
-    where
-    q<n : n / base < n
-    q<n = m/n<m n base (s<s z<s)
+  ... | false = (n % base) ∷ xs -- n here is already sufficient?
```

Comment:
What does "sufficient" in this comment mean?

Reply:
That in this case branch, IIUC, `n % base` is actually equal (propositionally) to `n`. But I'd be happier for a second pair of eyes to confirm that!

Reply:
It's partly a question of how these things get implemented later, I think, in the compiled haskell. If you can compute division with remainder in one go, then there's no point in relitigating the real tip ships between the various numerical values and their references. But on the face of it, it's an expensive computation of `n / base`, then a test for non-zeroness, and then an appeal to the value of the remainder `n % base`. 

A simpler analysis might be to test `does (n <? base)` and only compute `n % base` in the `false` branch of such a test. How much if this reflects the complex history of this module and it's friends, I don't really know, which perhaps meant I'd been (too?) conservative as to this refactoring. From the library point of view, I think the current definition is bonkers. But without really being clear about how this interacts with compilation via `MAlonzo` (I don't find it easy to discover anything about that pipeline), it's perhaps one to leave alone. But I (perhaps mistakenly) decided to leave the comment there for 'the future'...?

Reply:
Ah right in which case might a clearer comment be something along the lines of `-- Could this simply be `n :: xs`?`

Reply:
To be honest, I've never once used what comes out of Malonzo as a consideration for whether to make a refactoring or not. I'm not saying it's a bad thing to consider, but just a heads up that not considering it hasn't come to bite us at any point _yet_...

Reply:
Comment revised. 
As for MAlonzo my attempts to query how it behaves on the main issue tracker didn't yield any useful answer, so yes, happy not ever to have to think about it... ;-)